### DISCUSSION TITLE
Submarine upgrade costs

### DISCUSSION THREAD

**Discussion by Kulik242** [FakeFishGames/Barotrauma#4923] [Ideas and suggestions]
I think that the submarine upgrades should take into account its parameters like number of coilguns, loaders, hull segments, batteries, etc. For example the upgrade cost for junction boxes should be proportional to the number of junction boxes on the sub.

That's sort of a no-brainer, but wouldn't that put bigger subs on a big disadvantage? Should the cost rise linearly? The money making potential of big subs is the same as smaller subs. Transport missions could make a difference here (bigger subs culd carry more), what else?

**Comment by RadCatish** [comment]
I think that the earning potential of subs is mainly based on how many players there are, so maybe that could be a factor?
I think there are either too few upgrades, or they're too cheap - we have nearly all of them in our campaign and it feels like we're not that far through the campaign

**Reply by Kulik242** [reply]
You have a point there, more players don't necessarily earn more per mission, but they make the game much quicker which results in more money per hour played. So a twice as big sub with twice as much equipment should cost twice as much to upgrade?

I actually always thought that there are far too many upgrades. It's probably relative for how much you play. We manage two or three session a week and they are about 90 minutes each, i.e. two short missions or one long mission. I can see how we get a different perspective if you play more. 
Once you fully upgraded your sub, what incentive would you need to switch to a new sub and start upgrading it?

**Comment by NilanthAnimosus** [comment]
A simpler way to have this is either to make it set in the sub editor with a default, or translate prices based off the submarine cost itself (Or a multiplier in sub editor) and perhaps a few counts of related devices on board.

So having current costs based on a 10K sub cost, so a 20K submarine may cost twice as much to upgrade. a 100K one could be feeling like a lifetime however but their generally rare and few between and their huge.

Alternatively, a multiplier, so 1.0 being the default then a submarine could have a 1.25x upgrade cost, 2x upgrade cost, etc.

for either you choose above, you could finally have 2 turrets are cheaper to upgrade than 5 turrets, 15 power transfer components cheaper than 25.

However as a note missions give a static amount as reward as per their type usually, having more players and being able to do them faster as a large group doesn't mean you get more, and for those with less play time or less success in general (IE. 1-2 players) changing it to be far more grindy could be a bad idea.

In my short-lived campaigns I could never get many of them - too much was going into necessary supplies as nobody in my crew would even craft or deconstruct, and it was always a max diving team of 1, if 2 when I am lucky. picture doing this with a dugong that has no fabricators on board, with a high running cost and those upgrades can take a long time.

Wouldn't mind things balanced to scale though, still sounds like an ok to good idea.

**Reply by Kulik242** [reply]
I wouldn't let sub designers to have a say in upgrading prices. Seing like most custom subs have their purchase prices set way too low for how overpowered/big/overengineered they are (probably in an effort to make players play them). If the purchase price would be automatically set based on number of interactable components and their parameters (reactor power output, efficiency, battery capacity and recharge rate, etc.) , hull/wall segments and crew equipment that goes with the sub; then yes, upgrade cost could be based on that.

**Reply by NilanthAnimosus** [reply]
sub designers already have control over enough things (Like setting fuel usage to zero or no overloading, what the power usage is, engine speeds, time between repairs and rate of component decay if any at all etc) - the only reason I suggest it be there is if the sub creator wants to make a broken submarine over a balanced one - they have that already so its not exactly breaking an existing balance to just leave that freedom in there.

If sub creators can scale how well their systems work, and how well stocked they are, and how effectively cheaty or not they are, I don't see a reason to suddenly want to restrict the sub editor and tie its hands in how they choose to scale the submarines upgrade costs.

If you do wish to try to scale everything to be balanced though, the sub editor as a whole is fairly unbalanced and user-specified, I cant even picture how you would balance the purchase costs of said custom subs, the scaled pricing of values set in the editor, and other such. It would be nice to have upgrade costs based on amount of devices and existing capabilities, though every submarine is different and that would be equally as easy to bypass 'balanced style' by scaling all the values accordingly.

If you were to scale say, cost of devices based on power used / generated or power vs thrust ratio, a sub designer could simply tweak the values to decrease power usage and thus cost - and increase fuel usage just to add extra work to modify the upgrade cost.

I just don't personally see why a submarine creator should not have access to a scale for upgrade costs, vanilla submarines can use different values as well in a reasonable way, you can still scale them from capabilities and its still finally in the hands of the submarine creator how good/broken/unbalanced or balanced their submarine is. They are generally optional, and up to the server host or the player.

**Reply by Kulik242** [reply]
1. Editor being open to cheating isn't a reason to allow even more cheating.
2. It is not taking away from sub designers (being one myself), the feature isn't there in the first place.
3. If upgrading cost would relate to the number of components and properties, designers would find way to minmaxing but as you've said - if you use a high power engine you have to increase fuel consumption - at least some balancing would be necessary.
